eng/PublishSourceBuild.props

```diff
@@ -0,0 +1,119 @@
+<Project>
+  <!-- Add targets here that run logic that depends on the merged asset manifest and published packages. -->
+  <UsingTask TaskName="Microsoft.DotNet.SourceBuild.Tasks.LeakDetection.CheckForPoison" AssemblyFile="$(MicrosoftDotNetSourceBuildTasksLeakDetectionAssembly)" TaskFactory="TaskHostFactory" Condition="'$(EnablePoison)' == 'true'" />
+  <Target Name="ReportPoisonUsage"
+          AfterTargets="DiscoverArtifacts"
+          DependsOnTargets="CreatePrivateSourceBuiltArtifactsArchive"
+          Condition="'$(EnablePoison)' == 'true'"
+          Inputs="$(MSBuildProjectFullPath)"
+          Outputs="$(BaseIntermediateOutputPath)ReportPoisonUsage.complete">
+    <ItemGroup>
+      <!-- Include dotnet-sdk-*.tar.gz -->
+      <SdkArchive Include="$(ArtifactsAssetsDir)**/dotnet-sdk-*$(ArchiveExtension)" />
```

Comment:
Alternatively, this could be `@(SdkTarball)` to grab the SDK that we're about to publish (defined in `Publishing.props` in a very similar manner).

Reply:
This is post-publish though, right?

Reply:
This target is currently sequenced after the discovery of the artifacts that we will publish, but will run before publish. `SdkTarball` is initialized in the `DiscoverArtifacts` target that this runs after.

Reply:
Sorry, scratch that. This runs post-publish.

 However, the `ArtifactsAssetsDir` is pointing at the pre-publish directory. if you want to pull it from post-publish, you should pull from `SourceBuiltAssetsDir`.

Reply:
Given that they're all pulling from the pre-publish dirs, I think keeping it that way is the right thing to do for now?

Reply:
Sounds good.